go/src/koding/klient/machine/mount/sync/history/history_test.go

```diff
@@ -0,0 +1,74 @@
+package history_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"koding/klient/machine/index"
+	"koding/klient/machine/mount/sync/discard"
+	"koding/klient/machine/mount/sync/history"
+	"koding/klient/machine/mount/sync/synctest"
+)
+
+func TestHistory(t *testing.T) {
+	const historySize = 5
+
+	changes := []struct {
+		Change *index.Change
+		RecLen int
+	}{
+		{
+			Change: nil,
+			RecLen: 0,
+		},
+		{
+			Change: index.NewChange("a/", index.PriorityLow, 0),
+			RecLen: 3,
+		},
+		{
+			Change: index.NewChange("b/", index.PriorityMedium, 0),
+			RecLen: historySize,
+		},
+		{
+			Change: index.NewChange("c/", index.PriorityHigh, 0),
+			RecLen: historySize,
+		},
+	}
+
+	h := history.NewHistory(discard.NewDiscard(), historySize)
+
+	for i, change := range changes {
+		tLeft := time.Now()
+
+		if change.Change != nil {
+			if err := synctest.ExecChange(h, change.Change, time.Second); err != nil {
+				t.Fatalf("want err = nil; got %v (i:%d)", err, i)
+			}
+		}
+
+		recs := h.Get()
+		if len(recs) != change.RecLen {
+			t.Fatalf("want recs len = %d; got %d (i:%d)", change.RecLen, len(recs), i)
+		}
+
+		for j := len(recs) - 1; j >= 0 && (len(recs)-1-j) < 3; j-- {
```

Comment:
sorry but i didnt understand the logic here, why we care about the last 3 records? if 3 is important and a constant, we could simplify the loop